### PR TITLE
feat: data tx cli

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- Adds CLI methods for data msg server
- requests.go -> msgs.go
- update msgs.go to validate properly

Closes: #619 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/data/msgs.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func (m *MsgSignData) ValidateBasic() error {
+	for _, addr := range m.Signers {
+		if _, err := sdk.AccAddressFromBech32(addr); err != nil {
+			return sdkerrors.ErrInvalidAddress.Wrap(err.Error())
+		}
+	}
+	if m.Hash != nil {
+		return m.Hash.Validate()
+	}
+	return sdkerrors.ErrInvalidRequest.Wrap("hash cannot be empty")
```
Comment (ryanchristo) [inline x/data/msgs.go:36]:
```suggestion
	if m.Hash == nil {
		return sdkerrors.ErrInvalidRequest.Wrap("hash cannot be empty")
	}
	return m.Hash.Validate()
```

File: x/data/msgs_test.go [hunk 2]
```diff
@@ -85,13 +92,13 @@ func TestMsgSignDataRequest_GetSigners(t *testing.T) {
 	msg = &MsgSignData{Signers: nil}
 	require.Empty(t, msg.GetSigners())
 
-	msg = &MsgSignData{Signers: []string{"abcd"}}
-	require.Panics(t, func() {
-		msg.GetSigners()
-	})
+	msg = &MsgSignData{Signers: []string{""}}
+	require.Equal(t, "", msg.GetSigners()[0].String())
```
Comment (ryanchristo) [inline x/data/msgs_test.go:96]: I'm not sure I understand this change. Should we be checking for panic? What's the purpose of this check if we already check setting signers in lines 89-90?

**Review by ryanchristo — COMMENTED** [review]
Looking good. A few questions and suggestions.

File: x/data/msgs_test.go (hunk 2 above)
Comment (technicallyty) [reply x/data/msgs_test.go:96]: panics should never happen in GetSigners, as ValidateBasic is always called first. TBH, i think this test should just be removed, i dont think we have this type of test anywhere else.

File: x/data/client/testsuite/tx.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+func NewIntegrationTestSuite(cfg network.Config) *IntegrationTestSuite {
+	return &IntegrationTestSuite{cfg: cfg}
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	// We execute NewIntegrationTestSuite to set cfg field of IntegrationTestSuite
```
Comment (ryanchristo) [inline x/data/client/testsuite/tx.go:30]: `NewIntegrationTestSuite` is not called within this block. Is this comment still relevant?

Comment (technicallyty) [reply x/data/client/testsuite/tx.go]: ah good catch, removed it. looks like that comment is still in the group tx.go file (where i copied from), will remove that as well

File: x/data/client/tx.go [hunk 4]
```diff
@@ -74,87 +69,37 @@ func MsgAnchorDataCmd() *cobra.Command {
 // MsgSignDataCmd creates a CLI command for Msg/SignData.
 func MsgSignDataCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "sign [signer] [cid]",
-		Short: `Sign an arbitrary piece of data on the blockchain.`,
-		Args:  cobra.ExactArgs(2),
+		Use:     "sign [iri]",
+		Short:   `Sign a piece of on-chain data.`,
+		Long:    `Sign a piece of on-chain data, attesting to its validity. The data MUST be of graph type (rdf file extension).`,
+		Example: "sign regen:13toVgf5aZqSVSeJQv562xkkeoe3rr3bJWa29PHVKVf77VAkVMcDvVd.rdf",
```
Comment (ryanchristo) [inline x/data/client/tx.go:75]:
```suggestion
		Example: "regen tx data sign regen:13toVgf5aZqSVSeJQv562xkkeoe3rr3bJWa29PHVKVf77VAkVMcDvVd.rdf",
```

**Review by ryanchristo — APPROVED** [review]
tACK

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**